h/accounts/schemas.py

```diff
@@ -185,12 +185,16 @@ class RegisterSchema(CSRFSchema):
             colander.Length(
                 min=models.USERNAME_MIN_LENGTH,
                 max=models.USERNAME_MAX_LENGTH),
-            colander.Regex('(?i)^[A-Z0-9._]+$'),
+            colander.Regex(
+                '(?i)^[A-Z0-9._]+$',
+                msg=_("Must contain only letters, numbers, periods and "
+                      "underscores")),
             unique_username,
             unblacklisted_username,
         ),
         title=_('Username:'),
-        hint=_('between {min} and {max} characters').format(
+        hint=_('between {min} and {max} characters, containing only letters, '
+               'numbers, periods and underscores').format(
```

Comment:
This is shown next to the _Username:_ field on the form. I thought it would be good to tell the user what characters they're allowed to use before they come up with a username, not after.